src/town_gui.cpp

```diff
@@ -110,17 +104,15 @@ struct TownAuthorityWindow : Window {
 		if (buttons != displayed_actions_on_previous_painting) this->SetDirty();
 		displayed_actions_on_previous_painting = buttons;
 
-		this->vscroll->SetCount(numact + 1);
-
-		if (this->sel_index != -1 && !HasBit(buttons, this->sel_index)) {
```

Comment:
This block is important. For instance open town action window and select "transport rights", now keep the window open and disable "transport rights" in the setting, the line is removed (as expected) but still selected.
The issue is there are now 2 bitmasks, one for available actions and one for enabled actions, and this block was using the available mask and of course can't use the enabled mask because the meaning is different.

Reply:
Now deselecting the option if disabled in settings while window is open.